controller/environment_space_scope_blackbox_test.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/fabric8-services/fabric8-env/controller"
+	"github.com/fabric8-services/fabric8-env/gormapp"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+/*
+Setup:
+
+One space 	=> space1
+Three users	=> user1, user2, user3
+
+** user_space_scope matrix **
+================================
+user 	manage 	contribute 	view
+================================
+user1	yes		yes			yes
+user2	no		yes			yes
+user3	no		no			yes
+
+** user operation matrix **
+================================
+user 	create 	list		show
+================================
+user1	yes		yes			yes
+user2	no		yes			yes
+user3	no		no			no
+*/
+
+var testUser1 = &testauth.Identity{ID: uuid.NewV4(), Email: "user1@test.com", Username: "user1"} // user1
+var testUser2 = &testauth.Identity{ID: uuid.NewV4(), Email: "user2@test.com", Username: "user2"} // user2
+var testUser3 = &testauth.Identity{ID: uuid.NewV4(), Email: "user3@test.com", Username: "user3"} // user3
+
+type EnvironmentSpaceScopeSuite struct {
+	testsuite.DBTestSuite
+	db *gormapp.GormDB
+
+	svc        *goa.Service
+	ctrl       *controller.EnvironmentController
+	authServer *httptest.Server
+
+	ctx1      context.Context
+	ctx2      context.Context
+	ctx3      context.Context
+	spaceID   uuid.UUID // space1
+	publicKey *rsa.PublicKey
+}
+
+func TestEnvironmentSpaceScopeSuite(t *testing.T) {
+	config, err := configuration.New("")
+	require.NoError(t, err)
+	suite.Run(t, &EnvironmentSpaceScopeSuite{DBTestSuite: testsuite.NewDBTestSuite(config)})
+}
+
+func (s *EnvironmentSpaceScopeSuite) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+
+	s.db = gormapp.NewGormDB(s.DB)
+	s.authServer = s.startAuthServer()
+	authURL := "http://" + s.authServer.Listener.Addr().String() + "/"
+	authService, err := service.NewAuthService(authURL)
+	require.NoError(s.T(), err)
+
+	s.svc = testauth.UnsecuredService("enviroment-test")
+	s.ctrl = controller.NewEnvironmentController(s.svc, s.db, authService)
+	s.spaceID = uuid.NewV4()
+
+	s.ctx1, _, err = testauth.EmbedUserTokenInContext(context.Background(), testUser1)
+	require.NoError(s.T(), err)
+	s.ctx2, _, err = testauth.EmbedUserTokenInContext(context.Background(), testUser2)
+	require.NoError(s.T(), err)
+	s.ctx3, _, err = testauth.EmbedUserTokenInContext(context.Background(), testUser3)
+	require.NoError(s.T(), err)
+
+	tokenMgr, err := token.ReadManagerFromContext(s.ctx1)
+	require.NoError(s.T(), err)
+	keys := tokenMgr.PublicKeys()
+	require.NotNil(s.T(), keys)
+	require.NotEmpty(s.T(), keys)
+	s.publicKey = keys[0]
+}
+
+func (s *EnvironmentSpaceScopeSuite) TearDownSuite() {
+	s.DBTestSuite.TearDownSuite()
+	s.authServer.Close()
+}
+
+func (s *EnvironmentSpaceScopeSuite) TestAllScope() {
+	payload := newCreateEnvironmentPayload("osio-stage", "stage", "cluster1.com")
+	var newEnv *app.EnvironmentSingle
+
+	s.T().Run("user1_create", func(t *testing.T) {
```

Comment:
you could wrap all `user1` tests into an intermediate subtest, for even better clarity (but that's just a suggestion)

Reply:
@xcoulon how ?? can you point some example OR just write some pseudo code here.

Reply:
something like this:
```
s.T().Run("user1", func(t *testing.T) {
  t.Run("create", func(t *testing.T) {
    // your test here...
  })
})
```

Reply:
done [27d56dc](https://github.com/fabric8-services/fabric8-env/pull/14/commits/27d56dca488c91b4b8c047928fa988be027511a8)